pageserver/src/l0_flush.rs

```diff
@@ -2,13 +2,23 @@ use std::{num::NonZeroUsize, sync::Arc};
 
 use crate::tenant::ephemeral_file;
 
-#[derive(Default, Debug, PartialEq, Eq, Clone, serde::Deserialize)]
+#[derive(Debug, PartialEq, Eq, Clone, serde::Deserialize)]
 #[serde(tag = "mode", rename_all = "kebab-case", deny_unknown_fields)]
 pub enum L0FlushConfig {
-    #[default]
     PageCached,
     #[serde(rename_all = "snake_case")]
-    Direct { max_concurrency: NonZeroUsize },
+    Direct {
+        max_concurrency: NonZeroUsize,
+    },
+}
+
+impl Default for L0FlushConfig {
+    fn default() -> Self {
+        Self::Direct {
+            // TODO: using num_cpus results in different peak memory usage on different instance types.
+            max_concurrency: NonZeroUsize::new(usize::max(1, num_cpus::get())).unwrap(),
```

Comment:
Why not use the single constant we already have with the runtimes?

We have `num_cpus` only because it is used to track what tokio does (in a hotfix PR I did long ago). `std` now has [`std::thread::available_parallelism`](https://doc.rust-lang.org/stable/std/thread/fn.available_parallelism.html) which would be preferable.

The constant I am thinking of is: `crate::task_mgr::TOKIO_WORKER_THREADS`... Aah I see it got simplified to be the source of truth instead of tracking tokio.

Reply:
😇

Reply:
No objections to a follow-up PR that cleans up the usage of `num_cpus` from the entire pageserver code base.

Reply:
(Depending on how we run pageserver in the future (containerized settings), I think something more under our control might make sense. Don't want to spend cycles thinking about that at this time)